test/controllers/detailedvaluationrequest/DvrControllerSpec.scala

```diff
@@ -69,6 +67,132 @@ class DvrControllerSpec extends VoaPropertyLinkingSpec {
+    val caseRef = "234234"
+    val authorisationId = 4222211L
+    val uarn = 123123
+    val localAuthRef = "1234341234"
+
+    val testPropertyHistory =
+      propertyHistory.copy(uarn = uarn, localAuthorityReference = localAuthRef, addressFull = addressLine)
+
+    when(mockVmvConnector.getPropertyHistory(any())(any()))
+      .thenReturn(Future.successful(testPropertyHistory))
+
+    def resultCanChallenge(isOwner: Boolean) =
+      controller.canChallenge(
+        plSubmissionId = plSubmissionId,
+        assessmentRef = assessmentRef,
+        caseRef = caseRef,
+        authorisationId = authorisationId,
+        uarn = uarn,
+        isOwner = isOwner
+      )(request)
+  }
+
+  "can Challenge for IP" must "redirect to challenge advice page when canChallenge return None" in new CanChallengeSetup {
+
+    when(mockPropertyLinkConnector.canChallenge(any(), any(), any(), any())(any()))
+      .thenReturn(Future.successful(None))
+
+    val result = resultCanChallenge(true)
+
+    val urlBack =
+      s"http://localhost:9523/business-rates-property-linking/my-organisation/property-link/$plSubmissionId/valuations/$assessmentRef?uarn=$uarn"
+    val expectedRedirect =
+      s"http://localhost:9537/business-rates-valuation/property-link/valuations/startChallenge?backLinkUrl=$urlBack"
+
+    status(result) mustBe SEE_OTHER
+    redirectLocation(result) mustBe Some(expectedRedirect)
+  }
+
+  "can Challenge for Agent" must "redirect to challenge advice page when canChallenge return None" in new CanChallengeSetup {
+
+    when(mockPropertyLinkConnector.canChallenge(any(), any(), any(), any())(any()))
+      .thenReturn(Future.successful(None))
+
+    val result = resultCanChallenge(false)
+
+    val urlBack =
+      s"http://localhost:9523/business-rates-property-linking/my-organisation/property-link/clients/all/$plSubmissionId/valuations/$assessmentRef?uarn=$uarn"
+    val expectedRedirect =
+      s"http://localhost:9537/business-rates-valuation/property-link/valuations/startChallenge?backLinkUrl=$urlBack"
+
+    status(result) mustBe SEE_OTHER
+    redirectLocation(result) mustBe Some(expectedRedirect)
+  }
+
+  "can Challenge for IP" must "redirect to start challenge when canChallenge response is true" in new CanChallengeSetup {
+
+    val testCanChallengeResponse = canChallengeResponse.copy(result = true)
+
+    when(mockPropertyLinkConnector.canChallenge(any(), any(), any(), any())(any()))
+      .thenReturn(Future.successful(Some(testCanChallengeResponse)))
+
+    val result = resultCanChallenge(true)
+
+    val expectedRedirect =
+      s"http://localhost:9531/business-rates-challenge/property-link/$plSubmissionId/valuation/$assessmentRef/check/$caseRef/party/client/start?isDvr=true"
+
+    status(result) mustBe SEE_OTHER
+    redirectLocation(result) mustBe Some(expectedRedirect)
+  }
+
+  "can Challenge for Agent" must "redirect to start challenge when canChallenge response is true" in new CanChallengeSetup {
```

Comment:
I would prefer `redirect to start challenge when the user can challenge` instead of `redirect to start challenge when canChallenge response is true` but it's really a minor thing. No one really ever reads this so I think it's ok to leave alone. Just for future tests, try to express what the test does in non-implementation-specific terms